app/voip/server/lib/Helper.ts

```diff
@@ -0,0 +1,172 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+
+
+import { ILivechatVisitor } from '../../../../definition/ILivechatVisitor';
+import { VoipRoom } from '../../../models/server/raw';
+import { settings } from '../../../settings/server';
+import { Subscriptions, Users } from '../../../models/server';
+import { Logger } from '../../../../server/lib/logger/Logger';
+
+const logger = new Logger('VoipHelper');
+
+
+const emailValidationRegex = /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+export function validateEmail(email: string): boolean {
+	if (!emailValidationRegex.test(email)) {
+		throw new Meteor.Error('error-invalid-email', `Invalid email ${ email }`, { function: 'Voip.validateEmail', email });
+	}
+	return true;
+}
+
+/**
+ * How to deal with custom field? Commenting it for timebeibng
+ */
+/*
+export const parseAgentCustomFields = (customFields: any): any => {
+	if (!customFields) {
+		return;
+	}
+
+	const externalCustomFields = (): any [] => {
+		const accountCustomFields = settings.get('Accounts_CustomFields');
+		if (!accountCustomFields || (accountCustomFields as string).trim() === '') {
+			return [];
+		}
+
+		try {
+			const parseCustomFields = JSON.parse(accountCustomFields as string);
+			return Object.keys(parseCustomFields)
+				.filter((customFieldKey) => parseCustomFields[customFieldKey].sendToIntegrations === true);
+		} catch (error: any) {
+			Voip.logger.error(error);
+			return [];
+		}
+	};
+
+	const externalCF = externalCustomFields();
+	return Object.keys(customFields).reduce((newObj as Object, key) => (externalCF.includes(key) ? { ...newObj, [key]: customFields[key] } : newObj), null);
+};
+*/
+
+export const normalizeAgent = (agentId: string): any => {
+	if (!agentId) {
+		return;
+	}
+
+	if (!settings.get('Livechat_show_agent_info')) {
+		return { hiddenInfo: true };
+	}
+
+	const agent = Users.getAgentInfo(agentId);
+	const { customFields: agentCustomFields, ...extraData } = agent;
+
+	return Object.assign(extraData);
+};
+
+export const createVoipRoom = async (rid: string, name: string, agent: any, guest: ILivechatVisitor): Promise<any> => {
```

Comment:
There is actually some difference there. In case of Livechatroom, we send events and messages in case of new room creation for livechat. 
Though we can put that code conditionally too.